codegen/README.md

```diff
@@ -0,0 +1,394 @@
+# Mir Code Generator
+
+## Motivation
+
+Mir requires a lot of boilerplate code to be written, which is further exacerbated  when using the Domain Specific Language (DSL). This presents several issues:
+
+- Writing boilerplate is tedious and time-consuming, making Mir less attractive for prototyping, which is, arguably, one of its main applications.
+- It makes it difficult to evolve Mir, as a small change to the core or DSL may require manual changes to a large amount of existing boilerplate code.
+- It forces the [“waterfall”](https://en.wikipedia.org/wiki/Waterfall_model) model on the development of modules. Since any change to the event and message descriptions (currently, in `.proto` files) needs to be propagated to all the related boilerplate, the programmer is discouraged from making such changes and, thus, is incentivized to think through the implementation and decide which events and network messages will be used before they start actually implementing the logic of the module. This approach is not always optimal for prototyping and exploring new ideas.
+
+The code generator aims to address these problems and make Mir a higher level framework. It may also simplify migration to a different model for representing events and network messages, or even allow multiple such models to coexist by performing an automatic conversion between them when necessary.
+
+## Usage
+
+### Annotating the `.proto` definitions
+
+See `protos/mir/codegen_extensions.proto`, `protos/dsl/codegen_extensions.proto`, and `protos/net/codegen_extensions.proto` for the list of extensions.
```

Comment:
`protos/dsl/codegen_extensions.proto` seems to not be there. (Also mentioned at other places in this file.)

Reply:
Ah, thank you. This file used to exist but was removed. Fixed